packages/core/src/sync-realtime/service.ts

```diff
@@ -338,16 +341,18 @@ export const handleBlock = async (
 
   const shouldRequestLogs =
     positiveBloomFilter &&
-    (service.logFilterSources.length > 0 || service.factorySources.length > 0);
-  const shouldRequestTraces = service.callTraceSources.length > 0;
+    (service.logFilterSources.length > 0 ||
+      service.factoryLogSources.length > 0);
```

Comment:
Correct me if I'm wrong but I think this is redundant, and we can simply rename positiveBloomFilter to `shouldRequestLogs`?